[exporter/dynatrace] add multi-instance deployment note to README.md (#2009)
dynatrace-oss-contrib/opentelemetry-collector-contrib: exporter/dynatraceexporter/README.md

All configurations are optional, but if an `endpoint` other than the OneAgent metric ingestion endpoint is specified then an `api_token` is required.
To see all available options, see [Advanced Configuration](#advanced-configuration) below.

> > When using this exporter, it is RECOMMENDED to configure your SDKs to export DELTA metrics. 

Reviewer: ```suggestion
> When using this exporter, it is strongly RECOMMENDED to configure your SDKs to export metrics with DELTA temporality. 
```

> > When using this exporter, it is strongly RECOMMENDED to configure your SDKs to export metrics

Author: Done in [6c679f7](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/6c679f788806b4f65c89cc45fdbb6e31eadceb2c) :slightly_smiling_face:

> > When receiving CUMULATIVE metrics, this exporter will perform CUMULATIVE to DELTA conversions. 

Reviewer: ```suggestion
> When receiving CUMULATIVE metrics, this exporter will perform CUMULATIVE to DELTA conversion. 
```

Author: Applied in [25bd4c1](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/25bd4c123a1d57fb9e5fcb11ed0c7545e419be4a). Thanks. :slightly_smiling_face:

> > In a multiple-instance deployment of the OpenTelemetry Collector this conversion will produce

Reviewer: ```suggestion
> In a multiple-instance deployment of the OpenTelemetry Collector, this conversion can produce
```

> > OpenTelemetry Collector, this conversion can produce inconsistent data.

Author: I updated this to read 
> unless it can be guaranteed that metrics from the same source are processed by the same collector instance.

Please let me know what you think :slightly_smiling_face:

All configurations are optional, but if an `endpoint` other than the OneAgent metric ingestion endpoint is specified then an `api_token` is required.
To see all available options, see [Advanced Configuration](#advanced-configuration) below.

> > When using this exporter, it is strongly RECOMMENDED to configure the OpenTelemetry SDKs to export metrics

Author: Done in [c3b8e6b](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/c3b8e6bb082d989dd5724ab44921ba5a897ee4c4) :slightly_smiling_face:

All configurations are optional, but if an `endpoint` other than the OneAgent metric ingestion endpoint is specified then an `api_token` is required.
To see all available options, see [Advanced Configuration](#advanced-configuration) below.

> When using this exporter, it is strongly RECOMMENDED to configure the OpenTelemetry SDKs to export metrics
> with DELTA temporality. When receiving Sum or Histogram metrics with CUMULATIVE temporality, this exporter 
> will perform CUMULATIVE to DELTA conversion, which will drop the first received data point, as there is no

> > previous data point to compare it to. In a multiple-instance deployment of the OpenTelemetry Collector,

Author: No worries. I think it is important that this information is easily digestible as well :slightly_smiling_face: 
I applied the suggestions and I think it is a bit clearer now. [172e492](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/172e492c03de977877a4151627d8fe1764dbb527)

### tags (Deprecated, Optional)

**Deprecated: Please use [default_dimensions](#default_dimensions-optional) instead**

> # Limitations

Reviewer 2: Limitations alone sounds a bit pejorative.. in reality, it's limitation when using cumulatives. Maybe this is a bit better? 
`Limitations when using Cumulatives values`. Or `Particularities when exporting cumulative values`? Naming is so hard..

Author: Yep that is a difficult one. But I think it is a necessary eye-catcher for users to see that exporting cumulatives can lead to issues. I feel like using too soft wording could make readers glance over it.

Reviewer 2: Considerations sounds good. Maybe even `Considerations with cumulative values`?

Author: Considerations it is, thanks. :)

## Cumulative Data Points

When receiving Sum or Histogram metrics with CUMULATIVE temporality, this exporter

> needs to perform CUMULATIVE to DELTA conversion. This conversion can lead to missing

Reviewer 2: ```suggestion
performs CUMULATIVE to DELTA conversion. This conversion can lead to missing
```

Author: See [65a6e5e](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/65a6e5e2ac590a033a5789cb2663a7c230ad5030)

> or inconsistent data, as described below.

Reviewer 2: ```suggestion
or inconsistent data, as described below:
```

Author: See [65a6e5e](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/65a6e5e2ac590a033a5789cb2663a7c230ad5030)

### First Data Points are dropped

> Due to the conversion the exporter will drop the first received data point,

Reviewer 2: ```suggestion
Due to the conversion, the exporter will drop the first received data point,
```

Author: See [65a6e5e](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/65a6e5e2ac590a033a5789cb2663a7c230ad5030)

as there is no previous data point to compare it to. This can be circumvented by

> exporting DELTA values from the OpenTelemetry SDK.

Reviewer 2: ```suggestion
by configuring the OpenTelemetry SDK to export DELTA values
```
Maybe?

Author: See [65a6e5e](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/65a6e5e2ac590a033a5789cb2663a7c230ad5030)

> ### Multi-instance deployment

Reviewer 2: ```suggestion
### Multi-instance collector deployments
```

Author: See [65a6e5e](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/65a6e5e2ac590a033a5789cb2663a7c230ad5030)

In a multiple-instance deployment of the OpenTelemetry Collector, the conversion

> can produce inconsistent data unless it can be guaranteed that metrics from the

Reviewer 2: ```suggestion
can produce inconsistent data, unless it can be guaranteed that metrics from the
```

Author: Hmm, am I allowed to put a comma there? :sweat_smile:

Reviewer 2: Yeah I guess not, too eager on commas 😅

same source are processed by the same collector instance. This can be circumvented

> by exporting DELTA values from the OpenTelemetry SDK.

Reviewer 2: ```suggestion
by configuring the OpenTelemetry SDK to export DELTA values
```
Maybe?

Author: See [65a6e5e](https://github.com/dynatrace-oss-contrib/opentelemetry-collector-contrib/pull/2009/commits/65a6e5e2ac590a033a5789cb2663a7c230ad5030)
